### PR TITLE
Small fixes

### PR DESCRIPTION
**Pull request by aakselrod**
This PR does the following:

 * Demotes a normally benign log message in the `wallet` package
 * Fixes a race condition
 * Gets the best block from the correct source when `bitcoind` `Rescan()` is called, and continues the rescan until the best block even if new blocks are being generated during the rescan
 * Fixes some edge cases in notifications and makes the `bitcoind` back-end not notify its client wallet until notifications are requested

### CODE REVIEW

File: chain/bitcoind.go [hunk 1]
```diff
@@ -523,11 +537,15 @@ func (c *BitcoindClient) socketHandler(zmqClient *gozmq.Conn) {
...
 	// Get initial conditions.
-	bs, err := c.BlockStamp()
+	bestHash, bestHeight, err := c.GetBestBlock()
 	if err != nil {
 		log.Error(err)
 		return
 	}
+	bs := &waddrmgr.BlockStamp{
+		Height: bestHeight,
+		Hash:   *bestHash,
```
Comment (Roasbeef) [inline chain/bitcoind.go]: Should we also be setting the timestamp here?

File: chain/bitcoind.go [hunk 2]
```diff
@@ -916,8 +936,13 @@ func (c *BitcoindClient) filterBlock(block *wire.MsgBlock, height int32,
...
 
-	log.Debugf("Filtering block %d (%s) with %d transactions", height,
-		block.BlockHash(), len(block.Transactions))
+	// Only mention that we're filtering a block if the client wallet has
+	// started monitoring the chain.
+	if !c.notifying() {
+		log.Debugf("Filtering block %d (%s) with %d transactions",
+			height, block.BlockHash(), len(block.Transactions))
+	}
+
```
Comment (Roasbeef) [inline chain/bitcoind.go:982]: Ah or seems fine as it's set here below?

File: chain/bitcoind.go [hunk 3]
```diff
@@ -919,6 +930,22 @@ func (c *BitcoindClient) rescan(hash *chainhash.Hash) error {
 		if i%10000 == 0 {
 			c.onRescanProgress(lastHash, i, block.Header.Timestamp)
 		}
+
+		if i == bestBlock.Height {
```
Comment (Roasbeef) [inline chain/bitcoind.go]: Why was this needed? Could use a comment explaining rationale.

File: chain/bitcoind.go (hunk 1 above)
Comment (aakselrod) [reply chain/bitcoind.go]: As you said, it's set below for subsequent blocks, but I've gone ahead and added the timestamp setting here as well for future use.

File: chain/bitcoind.go (hunk 2 above)
Comment (aakselrod) [reply chain/bitcoind.go:982]: Yes, but I've gone ahead and added that above as well.

File: chain/bitcoind.go (hunk 3 above)
Comment (aakselrod) [reply chain/bitcoind.go]: Added comment.

**Review by Roasbeef — APPROVED** [review]
LGTM 💫

**Review by Roasbeef — APPROVED** [review]
LGTM 🍿
